x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_embeddable.tsx

```diff
@@ -45,12 +46,56 @@ export class SingleMetricViewerEmbeddable extends Embeddable<
   private reload$ = new Subject<void>();
   public readonly type: string = ANOMALY_SINGLE_METRIC_VIEWER_EMBEDDABLE_TYPE;
 
+  // API
+  public readonly functionDescription: BehaviorSubject<string | undefined>;
+  public readonly jobId: BehaviorSubject<JobId | undefined>;
+  public readonly selectedDetectorIndex: BehaviorSubject<number | undefined>;
+  public readonly selectedEntities: BehaviorSubject<MlEntityField[] | undefined>;
+  public readonly title: BehaviorSubject<string | undefined>;
+
+  private apiSubscriptions = new Subscription();
+
   constructor(
     initialInput: SingleMetricViewerEmbeddableInput,
     public services: [CoreStart, MlDependencies, SingleMetricViewerServices],
     parent?: IContainer
   ) {
     super(initialInput, {} as AnomalyChartsEmbeddableOutput, parent);
+
+    this.jobId = embeddableInputToSubject<JobId, SingleMetricViewerEmbeddableInput>(
+      this.apiSubscriptions,
+      this,
+      'jobId'
+    );
+
+    this.functionDescription = embeddableInputToSubject<
+      string | undefined,
+      SingleMetricViewerEmbeddableInput
+    >(this.apiSubscriptions, this, 'functionDescription');
+
+    this.selectedDetectorIndex = embeddableInputToSubject<
+      number | undefined,
+      SingleMetricViewerEmbeddableInput
+    >(this.apiSubscriptions, this, 'selectedDetectorIndex');
+
+    this.selectedEntities = embeddableInputToSubject<
+      MlEntityField[] | undefined,
+      SingleMetricViewerEmbeddableInput
+    >(this.apiSubscriptions, this, 'selectedEntities');
+
+    this.title = embeddableInputToSubject<string, SingleMetricViewerEmbeddableInput>(
+      this.apiSubscriptions,
+      this,
+      'title'
+    );
```

Comment:
`title` is a common prop and is handled by the embeddable framework

```suggestion
```

Reply:
Yes, it can be handled via `Panel settings` but is also editable when using the `Edit panel` action that has been added. 
When converting to the new framework, this will be removed, but right now it's still handled by the edit action. 
I will remove this when refactoring to the new framework in a follow up PR.

Reply:
Anomaly swim lane has exactly the same - editable via `Panel settings` and `Edit panel` actions, but it does not duplicate `title` API.

Reply:
Good call - removed in c39b883e6b82023461799d2c4ab78623359a1011